src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/SvType.java

```diff
@@ -1,41 +1,166 @@
 package org.broadinstitute.hellbender.tools.spark.sv.discovery;
 
 import htsjdk.variant.variantcontext.Allele;
+import htsjdk.variant.variantcontext.VariantContextBuilder;
+import htsjdk.variant.vcf.VCFConstants;
+import org.apache.commons.lang3.EnumUtils;
+import org.broadinstitute.hellbender.engine.datasources.ReferenceMultiSource;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.tools.spark.sv.discovery.inference.NovelAdjacencyAndAltHaplotype;
+import org.broadinstitute.hellbender.tools.spark.sv.utils.GATKSVVCFConstants;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
 
-import java.util.Collections;
-import java.util.Map;
+import java.io.IOException;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static org.broadinstitute.hellbender.tools.spark.sv.utils.GATKSVVCFConstants.INTERVAL_VARIANT_ID_FIELD_SEPARATOR;
 
 
 /**
- * Various types of structural variations
+ * Various types of structural variations.
+ * Holding minimum information on VariantContext
+ * it can hold:
+ * CHR, POS, ID, REF, ALT, SVTYPE, END (optional), SVLEN(optional)
  */
 public abstract class SvType {
+    public static final int NO_APPLICABLE_END = -1;
+    public static final int NO_APPLICABLE_LEN = -1;
 
+    protected final String variantCHR;
+    protected final int variantPOS;
+    protected final int variantEND;
     protected final String variantId;
+    protected final Allele refAllele;
     protected final Allele altAllele;
     protected final int svLen;
-    protected final Map<String, String> extraAttributes;
+    protected final Map<String, Object> typeSpecificAttributes;
 
-    protected static final Map<String, String> noExtraAttributes = Collections.emptyMap();
+    protected static final Map<String, Object> noExtraAttributes = Collections.emptyMap();
 
-    protected SvType(final String id, final Allele altAllele, final int len, final Map<String, String> typeSpecificExtraAttributes) {
-        variantId = id;
+    public SvType(final String variantCHR, final int variantPOS, final int variantEND, final String variantId,
+                  final Allele refAllele, final Allele altAllele, final int svLen, final Map<String, Object> typeSpecificAttributes) {
+        this.variantCHR = variantCHR;
```

Comment:
I'm not sure about the capitalization of these fields and parameters.. why not `variantChromosome`, `variantPosition`, etc? Seems inconsistent.

Reply:
renamed to be the same as methods to construct VariantContextBuilder and documented